src/shogun/clustering/KMeans.cpp

```diff
@@ -71,7 +71,8 @@ void CKMeans::Lloyd_KMeans(SGMatrix<float64_t> centers, int32_t num_centers)
 				   	Terminating. \n", iter)
 
 		changed=0;
-		rhs_mus->copy_feature_matrix(centers);
+		rhs_mus->set_feature_matrix(centers.clone());
```

Comment:
No clue. Without the clone, one of the unit-test fails. So kept the behavior just the same as before. Maybe Saurabh can fill this in.